extensions/shared/components/upgrade-nudge/index.jsx

```diff
@@ -90,13 +102,19 @@ export default compose( [
 				redirect_to,
 			} );
 
+		const planName = get( plan, [ 'product_name' ] );
 		return {
-			trackEvent: blockName =>
+			trackViewEvent: () =>
+				void analytics.tracks.recordEvent( 'jetpack_editor_block_upgrade_nudge_impression', {
+					planName,
+					block: blockName,
+				} ),
+			trackClickEvent: () =>
 				void analytics.tracks.recordEvent( 'jetpack_editor_block_upgrade_click', {
-					plan,
+					planName,
```

Comment:
I think we should keep the same property name as before `plan: planName`.

Also note that event properties should be in `snake_case`. 🙂